### ISSUE TITLE
Float window cannot be split

### ISSUE THREAD

**Issue by Amleto** [NeogitOrg/neogit#1364]
### Description

If commit_editor.kind is set to "floating", when I try to commit I get an exception that floating window cannot be split.

### Neovim version

v0.10.0
Release

### Operating system and version

Win11/wsl Ubuntu 20.04

### Steps to reproduce

Stage something then 'cc' to trigger the commit editor

### Expected behavior

Float window opens with no exceptions

### Actual behavior

Exception thrown because diffs are attempted to be displayed by splitting the float window (but float windows can't be split, apparently)

### Minimal config

```Lua
-- NOTE: See the end of this file if you are reporting an issue, etc. Ignore all the "scary" functions up top, those are
-- used for setup and other operations.
local M = {}

local base_root_path = vim.fn.fnamemodify(debug.getinfo(1, "S").source:sub(2), ":p:h") .. "/.min"
function M.root(path)
  return base_root_path .. "/" .. (path or "")
end

function M.load_plugin(plugin_name, plugin_url)
  local package_root = M.root("plugins/")
  local install_destination = package_root .. plugin_name
  vim.opt.runtimepath:append(install_destination)

  if not vim.loop.fs_stat(package_root) then
    vim.fn.mkdir(package_root, "p")
  end

  if not vim.loop.fs_stat(install_destination) then
    print(string.format("> Downloading plugin '%s' to '%s'", plugin_name, install_destination))
    vim.fn.system({
      "git",
      "clone",
      "--depth=1",
      plugin_url,
      install_destination,
    })
    if vim.v.shell_error > 0 then
      error(string.format("> Failed to clone plugin: '%s' in '%s'!", plugin_name, install_destination),
        vim.log.levels.ERROR)
    end
  end
end

---@alias PluginName string The plugin name, will be used as part of the git clone destination
---@alias PluginUrl string The git url at which a plugin is located, can be a path. See https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols for details
---@alias MinPlugins table<PluginName, PluginUrl>

---Do the initial setup. Downloads plugins, ensures the minimal init does not pollute the filesystem by keeping
---everything self contained to the CWD of the minimal init file. Run prior to running tests, reproducing issues, etc.
---@param plugins? table<PluginName, PluginUrl>
function M.setup(plugins)
  vim.opt.packpath = {}                      -- Empty the package path so we use only the plugins specified
  vim.opt.runtimepath:append(M.root(".min")) -- Ensure the runtime detects the root min dir

  -- Install required plugins
  if plugins ~= nil then
    for plugin_name, plugin_url in pairs(plugins) do
      M.load_plugin(plugin_name, plugin_url)
    end
  end

  vim.env.XDG_CONFIG_HOME = M.root("xdg/config")
  vim.env.XDG_DATA_HOME = M.root("xdg/data")
  vim.env.XDG_STATE_HOME = M.root("xdg/state")
  vim.env.XDG_CACHE_HOME = M.root("xdg/cache")

  -- NOTE: Cleanup the xdg cache on exit so new runs of the minimal init doesn't share any previous state, e.g. shada
  vim.api.nvim_create_autocmd("VimLeave", {
    callback = function()
      vim.fn.system({
        "rm",
        "-r",
        "-f",
        M.root("xdg")
      })
    end
  })
end

-- NOTE: If you have additional plugins you need to install to reproduce your issue, include them in the plugins
-- table within the setup call below.
M.setup({
  plenary = "https://github.com/nvim-lua/plenary.nvim.git",
  telescope = "https://github.com/nvim-telescope/telescope.nvim",
  diffview = "https://github.com/sindrets/diffview.nvim",
  neogit = "https://github.com/NeogitOrg/neogit"
})
-- WARN: Do all plugin setup, test runs, reproductions, etc. AFTER calling setup with a list of plugins!
-- Basically, do all that stuff AFTER this line.
require("neogit").setup({commit_editor={kind="floating"}}) -- For instance, setup Neogit
```

**Comment by CKolkey** [comment]
Oh.. true. If you disable the staged diffs it wont try to split. TBH i don't recommend using floating windows - just use a tab. I don't want to ship a layout engine for managing floating windows like telescope 😅 too many other things that are higher impact/interest

**Comment by acampove** [comment]
@CKolkey And how do we disable the staged diffs? I am seeing the same problem:

![image](https://github.com/user-attachments/assets/b244ddc2-e239-4ecb-93d4-bdf1183599ea)

**Comment by CKolkey** [comment]
If you have the commit editor set to "floating", the diff split will never be shown: https://github.com/NeogitOrg/neogit/pull/1481

**Comment by acampove** [comment]
@CKolkey And how do we set the commit editor not to be "floating"?

**Comment by CKolkey** [comment]
Oh, it's not by default. Maybe you have something in your config that is doing that?

**Comment by apyrkh** [comment]
I have the same issue caused by the following configuration:
```
    commit_editor = {
      kind = "vsplit",
    },
```

When there is pre-commit hook, which takes more than `console_timeout` (`2000` by default) - neogit console appears and after that it causes the error
@CKolkey

**Comment by acampove** [comment]
> I have the same issue caused by the following configuration:
> 
> ```
>     commit_editor = {
>       kind = "vsplit",
>     },
> ```
> 
> When there is pre-commit hook, which takes more than `console_timeout` (`2000` by default) - neogit console appears and after that it causes the error @CKolkey

I do not have that, in my case it is:

```lua
        require("neogit").setup(
            {
                commit_editor={
                    kind="split"
                }
            })
```
